lib/remote/actionshandler.cpp

```diff
@@ -88,7 +89,28 @@ bool ActionsHandler::HandleRequest(
 	if (params)
 		verbose = HttpUtility::GetLastParameter(params, "verbose");
 
+	std::shared_lock wgLock{*waitGroup, std::try_to_lock};
+	if (!wgLock) {
+		HttpUtility::SendJsonError(response, params, 503, "Shutting down.");
+		return true;
+	}
+
 	for (ConfigObject::Ptr obj : objs) {
+		if (!waitGroup->IsLockable()) {
+			if (wgLock) {
+				wgLock.unlock();
+			}
+			
```

Comment:
```suggestion

```

(There is trailing whitespace in this line, I'm not sure how clear this will be from what GitHub shows, I'm **not** asking to remove the empty line.)